File: backend/app/models/task.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from ..base import Base
import datetime


class Task(Base):
    __tablename__ = "tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    lesson_id = Column(Integer, ForeignKey("lessons.id"), nullable=False)
    is_completed = Column(Integer, default=0)  # 0 = incomplete, 1 = complete
    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    
    # Relationship to lesson
    lesson = relationship("Lesson", back_populates="tasks")

File: frontend/lib/domain/entities/task.dart
import 'package:equatable/equatable.dart';

class Task extends Equatable {
  final int id;
  final String title;
  final String description;
  final bool isCompleted;
  final int lessonId;

  const Task({
    required this.id,
    required this.title,
    required this.description,
    required this.isCompleted,
    required this.lessonId,
  });

  @override
  List<Object?> get props => [id, title, description, isCompleted];
}
